src/webapp/pages/settings/useConfig.ts

```diff
@@ -28,6 +31,14 @@ export function useConfig(): useConfigPloc {
         [compositionRoot]
     );
 
+    const updateGoogleAnalyticsCode = useCallback(
+        async (code: string) => {
+            setGoogleAnalyticsCode(code);
+            compositionRoot.config.updateGoogleAnalyticsCode(code);
```

Comment:
Add await here, if not in SettingsPage component, when you are saving the G4 code, the page reloads without waiting to save the code, so never is saved